bundle/Service/ScheduledVisibilityService.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Netgen\Bundle\IbexaScheduledVisibilityBundle\Service;
+
+use DateTime;
+use Ibexa\Contracts\Core\Repository\Values\Content\Content;
+use Ibexa\Contracts\Core\Repository\Values\Content\Field;
+use Ibexa\Core\FieldType\Date\Value as DateValue;
+use Ibexa\Core\FieldType\DateAndTime\Value as DateAndTimeValue;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\Enums\VisibilityUpdateResult;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\ScheduledVisibility\Registry;
+use OutOfBoundsException;
+
+final class ScheduledVisibilityService
+{
+    public function __construct(
+        private readonly Registry $registry,
+    ) {}
+
+    /**
+     * @throws OutOfBoundsException
+     */
+    public function updateVisibilityIfNeeded(Content $content, ?string $handlerIdentifier = null): VisibilityUpdateResult
+    {
+        if (!$this->accept($content)) {
+            return VisibilityUpdateResult::NoChange;
+        }
+
+        $handler = $this->registry->get($handlerIdentifier);
+
+        if ($this->shouldBeHidden($content) && !$handler->isHidden($content)) {
+            $handler->hide($content);
+
+            return VisibilityUpdateResult::Hidden;
+        }
+
+        if ($this->shouldBeVisible($content) && $handler->isHidden($content)) {
```

Comment:
This will not work correctly as `isHidden()` is not the negation of `isVisible`, see https://github.com/netgen/ibexa-scheduled-visibility/pull/1/files#r1651238044

Instead it should check:

```php
$this->shouldBeVisible($content) && !$handler->isVisible($content)
```